app-modules/notifications/src/Actions/SubscriptionCreate.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Assist\Notifications\Actions;
+
+use App\Models\User;
+use Assist\Notifications\Models\Contracts\Subscribable;
+use Illuminate\Database\UniqueConstraintViolationException;
+use Assist\Notifications\Exceptions\SubscriptionAlreadyExistsException;
+
+class SubscriptionCreate
+{
+    public function handle(User $user, Subscribable $subscribable, bool $throwUniqueException = true): void
+    {
+        try {
+            $user->subscriptions()->create([
+                'subscribable_id' => $subscribable->getKey(),
+                'subscribable_type' => $subscribable->getMorphClass(),
+            ]);
```

Comment:
Should we try to first find an existing but deleted instance of this relationship? Or is the intention with the subscription soft delete to have a more fine grained historical trace (beyond the audit) of exactly when a user might have subscribed/unsubscribed from a Student/Prospect?

Reply:
I debated this in my head for a while, and I landed on deleting them and creating new records because I think data may soon be stored on a Subscription.

For example, the `via` concept we discussed where we could set "why" they were subscribed (i.e., direct subscription, touching a related item, etc.)

And I figured that we would want to retain that data rather than overwrite it.

Reply:
Ahh yeah - that's a great point.